pkg/cache/cluster.go

```diff
@@ -596,23 +615,32 @@ func (c *clusterCache) sync() error {
 
 // EnsureSynced checks cache state and synchronizes it if necessary
 func (c *clusterCache) EnsureSynced() error {
-	// first check if cluster is synced *without lock*
+	syncStatus := c.syncStatus
+
+	// first check if cluster is synced *without acquiring the full clusterCache lock*
+	syncStatus.lock.Lock()
 	if c.synced() {
-		return c.syncError
+		syncError := syncStatus.syncError
+		syncStatus.lock.Unlock()
+		return syncError
 	}
+	syncStatus.lock.Unlock() // release the lock, so that we can acquire the parent lock (see struct comment re: lock acquisition ordering)
 
 	c.lock.Lock()
+	syncStatus.lock.Lock()
 	defer c.lock.Unlock()
+	defer syncStatus.lock.Unlock()
```

Comment:
```suggestion
	c.lock.Lock()
	defer c.lock.Unlock()
	syncStatus.lock.Lock()
	defer syncStatus.lock.Unlock()
```
Reads better and is more logical? Same in another method.

Reply:
Done (in both places)... my mild personal preference was to have it the other way, but they are semantically equivalent, and so I'm happy if you're happy 😄